File: experiments/source/python/affine_score_reduce.py
import loom


@loom.entry
def affine_score_reduce(scale: float, bias: float, threshold: float, x: loom.Tensor1) -> float:
    def score(xi: float) -> float:
        value = (scale * xi) + bias
        return value if value > threshold else 0.0

    return loom.Tensor1.reduce_sum(loom.Tensor1.map(score, x))

File: experiments/source/cpp/affine_score_reduce.cpp
#include <loom/loom.hpp>

LOOM_ENTRY float affine_score_reduce(float scale, float bias, float threshold, loom::Tensor1 x) {
  auto score = [=](float xi) -> float {
    auto value = (scale * xi) + bias;
    return value > threshold ? value : 0.0f;
  };
  return loom::Tensor1::reduce_sum(loom::Tensor1::map(score, x));
}
